Zend/zend_execute_API.c

```diff
@@ -1522,7 +1522,9 @@ static void zend_set_timeout_ex(zend_long seconds, bool reset_signals) /* {{{ */
 			t_r.it_value.tv_sec = seconds;
 			t_r.it_value.tv_usec = t_r.it_interval.tv_sec = t_r.it_interval.tv_usec = 0;
 
-# if defined(__CYGWIN__) || defined(__PASE__)
+# if defined(__CYGWIN__) || defined(__PASE__) || (defined(__aarch64__) && defined(__APPLE__))
+			// ITIMER_PROF is broken in Apple Silicon system with MacOS >= 14. The SIGPROF signal is sent way too early
+			// when the process opens sockets.
 			setitimer(ITIMER_REAL, &t_r, NULL);
 		}
 		signo = SIGALRM;
```

Comment:
``SIGALRM`` may conflict with ``sleep()``. I don't have Mac and can't check this.

Reply:
What do you have in mind exactly? I can check it on my machine. Do you think `sleep` is just broken?

Reply:
To prevent any conflict (for instance with `pcntl_alarm()`, you could use `SIGIO`, as in https://github.com/php/php-src/pull/13468#discussion_r1499698960.

Reply:
I meant calls to ``sleep()`` may conflict with signal handlers for ``SIGALARM``.
This is the old well known problem. (see ``man 3 sleep`` - "On some systems, sleep() may be implemented using alarm(2) and SIGALRM (POSIX.1 permits this); mixing calls to alarm(2) and sleep() is a bad idea.")

I can't check this PR, and I can't take a decision about its acceptance or rejection.

Reply:
@dstogov according to the MacOS man page, sleep() does not use SIGALARM on this platform:

> This function is implemented using nanosleep(2) by pausing for seconds seconds or until a signal occurs.  Consequently, in this implementation, sleeping has no effect on the state of process timers, and there is no special handling for SIGALRM.

Reply:
I can confirm @arnaud-lb. This is what the man-page says.